utils/docker-utils.sh

```diff
@@ -129,7 +137,95 @@ docker::prune_configs() {
-    docker config rm $(docker config ls -qf label=name="$CONFIG_LABEL") &>/dev/null
+    if [[ -n $(docker config ls -qf label=name="$CONFIG_LABEL") ]]; then
+        log info "Waiting for configs to be removed..."
+
+        docker config rm $(docker config ls -qf label=name="$CONFIG_LABEL") &>/dev/null
+
+        overwrite "Waiting for configs to be removed... Done"
+    fi
+}
+
+# Deploy a service, it will set config digests (in case a config is defined in the compose file)
+#
+# Arguments:
+# - $1 : docker compose path, e.g. "/instant/monitoring" ...
+# - $2 : docker compose file, e.g. "docker-compose.yml" "docker-compose.cluster.yml" ...
+# - $3 : docker compose dev file, e.g. "docker-compose.dev.yml"
+# - $4 : services names, e.g. "monitoring" "hapi-fhir" ...
+#
+docker::deploy_service() {
+    local -r DOCKER_COMPOSE_PATH="${1:?"FATAL: function 'deploy_service' is missing a parameter"}"
+    local -r DOCKER_COMPOSE_FILE="${2:?"FATAL: function 'deploy_service' is missing a parameter"}"
+    local -r DOCKER_COMPOSE_DEV_FILE="${3:-""}"
+    local docker_compose_dev=""
+
+    # Check for need to set config digests
+    local -r files=($(yq '.configs."*.*".file' "${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE"))
+    if [[ "${files[*]}" != "null" ]]; then
+        config::set_config_digests "${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE"
+    fi
+
+    if [[ -n "${DOCKER_COMPOSE_DEV_FILE}" ]]; then
+        docker_compose_dev="-c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_FILE"
+    fi
+
+    try "docker stack deploy \
+        -c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE \
+        $docker_compose_dev \
+         instant" \
+        throw \
+        "Wrong configuration in ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE or in $docker_compose_dev"
+
+    # Remove stale configs according to the labels in the compose file
+    local -r label_names=($(yq '.configs."*.*".labels.name' "${DOCKER_COMPOSE_PATH}/${DOCKER_COMPOSE_FILE}" | sort -u))
+    if [[ "${label_names[*]}" != "null" ]]; then
+        for label_name in "${label_names[@]}"; do
+            config::remove_stale_service_configs "$COMPOSE_FILE_PATH/$DOCKER_COMPOSE_FILE" "${label_name}"
+        done
+    fi
+}
+
+# Deploy a config importer:
+# Sets the config digests, deploy the config importer, remove it and remove the stale configs
+#
+# Arguments:
+# - $1 : docker compose path, e.g. "/instant/monitoring/importer/docker-compose.config.yml" ...
+# - $2 : services name, e.g. "clickhouse-config-importer" ...
+# - $3 : config label, e.g. "clickhouse" "kibana" ...
+# - $4 : docker compose file name, default "docker-compose.config.yml"
+#
+docker::deploy_config_importer() {
+    local -r CONFIG_COMPOSE_PATH="${1:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+    local -r SERVICE_NAME="${2:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+    local -r CONFIG_LABEL="${3:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
```

Comment:
Let's refactor the error message into a variable and then reuse. Maybe a function that would replace the token "deploy_config_importer"

Reply:
yes cool, I will consider updating this in another PR alone